CIPs/cip-137.md

```diff
@@ -18,190 +18,239 @@ An API for reading and writing events on Ceramic event streams.
+```
 
-- *range* <Array<string>> - a Recon message, representing the latest state of the range described by the subscription request. In the form of `[eventid, ahash, eventid]`
+#### Parameters:
 
----
+* *streamid* <string> - the stream to subscribe to (Normally `*` is used to subscribe to sets of streams)
 
+* *{sort-key}* <string> - the *sort-key* and *sort-value* to subscribe to (e.g. `model` and `kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr` respectively)
+* *ctrlRangeStart* <string> - hex encoded string describing the start of the range of controllers to sync (defaults to `0x0000000000000000`)
+* *ctrlRangeEnd* <string> - hex encoded string describing the end of the range of controllers to sync (defaults to `0xffffffffffffffff`)
+* *offset* <integer> - the number of eventId bytes already consumed
+* *limit* <integer> - the max number of events to return
+* *duration* <integer> - the amount of time, in seconds, to wait for a response
 
 
-### `ceramic_unsubscribe`
+#### Returns:
 
-This method tells the Ceramic node to unsubscribe from a given interest range.
+There are two different content types supported as return value, as specified by the [`Accept`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) http header.
 
-**Params:**
+**JSON:**
 
-- *ceramicUrl* <string> - the Ceramic URL to unsubscribe from
-- *options* <object> - options for the subscription
-  - *ctrlRangeStart* <string> - hex encoded string describing the start of the range of controllers to sync (defaults to `0x0000000000000000`)
-  - *ctrlRangeEnd* <string> - hex encoded string describing the end of the range of controllers to sync (defaults to `0xffffffffffffffff`)
+Content-Type:  `application/json`
 
-**Returns:**
+This content type returns a json representation of the events received. Note that this is just a representation of the event content, full verifiability is not retained.
 
-- *success* <boolean> - true if the request was successful
+- *events* <array<object>> - the list of events
+  - *eventId* <string> - the id of the event
+  - *id* <string> - CID of the InitEvent for this stream
+  - *prev* <array<string>> - CIDs of previous hash linked events
+  - *header* <object> - header for the event
+  - *data* <object> - the data of the event
+  - *timestamp* <integer> - the unixtime this event was timestamped (if it has been)
+- *offset* <integer> - the number of eventId bytes already consumed
 
----
 
 
+**CAR file:**
+
+Content-Type:  `application/vnd.ipld.car`
+
+This content types returns the underlaying IPLD data structure of the event streams encoded as a CAR file. This data can be used to trustlessly verify the integrity of the event stream. 
+
+The root object of the CAR file is a *SubscriptionResult* which contains an array of *EvenIds* the new *offset*. The CAR file also includes all *Event* objects, corresponding to the eventids in the subscription result.
```

Comment:
```suggestion
The root object of the CAR file is a *SubscriptionResult* which contains an array of *EventIds* the new *offset*. The CAR file also includes all *Event* objects, corresponding to the EventIds in the subscription result.
```

Reply:
"[...] which contains an array of EventIds the new offset." <- the flow of this sentence doesn't make sense.  What are we trying to say about the new offset?